platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -523,17 +541,19 @@ odp_ipsec_sa_t odp_ipsec_sa_create(const odp_ipsec_sa_param_t *param)
 
 	ipsec_sa->icv_len = crypto_param.auth_digest_len;
 
-	if (param->crypto.cipher_key_extra.length) {
-		if (param->crypto.cipher_key_extra.length >
-		    IPSEC_MAX_SALT_LEN)
+	if (ipsec_sa->salt_length) {
+		if (ipsec_sa->salt_length > IPSEC_MAX_SALT_LEN) {
+			ODP_ERR("IPSEC_MAX_SALT_LEN too small\n");
 			goto error;
+		}
```

Comment:
Argh. This needs to be fixed in master branch as well.

Reply:
Done in v2.